cfpb/consumerfinance.gov
# Future dates cause errors in filterable lists

This change should fix an internal server error seen recently in issue [5365](https://github.local/CFGOV/cfgov-alerts/issues/5365), that has something to do with how filterable lists handle future dates.
Putting future dates in the `from_date` and `to_date` filterable list fields should not cause an internal server error.


## Additions

- Add default_input_formats to `FilterableDateField` class in `cfgov/v1/forms.py`
- Add [freezegun](https://pypi.org/project/freezegun/) and test for future dates


## Removals

- `cfgov/v1/util/date_filter.py` module

## Changes

- `cfgov/v1/forms.py`
- `cfgov/v1/tests/test_forms.py`
- `cfgov/v1/tests/util/test_datetimes.py`
- `cfgov/v1/util/datetimes.py`

## How to test this PR

Attempt to add future date into the FilterableList form using a Safari or IE11 browser
e.g. from_date and to_date: 05/04/2021 and 11/04/2021
Try flipping the month and day, European style, which looks like a time range in the future
Filter a filterable list on a date range entirely in the future

1. Visit https://www.consumerfinance.gov/about-us/newsroom/ in Safari or IE11 in Sauce Labs, or any other browser that doesn't support the native date input control
2. Enter some date filter and click Apply Filters
3. Verify there is no internal server error


## Screenshots


## Notes and todos

- This must be testing in IE11, Safari, and Opera Mini browsers


## Checklist

  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### Front-end testing

#### Browser testing

Visually tested in the following supported browsers:



#### Accessibility


#### Other
